.github/workflows/dockerpublish.yml

```
@@ -22,87 +22,63 @@ env:
         run: echo "${{ secrets.DOCKER_HUB_PASSWORD }}" | docker login -u "${{ github.actor }}" --password-stdin
 
-      - name: Push latest image
-        run: |
-          IMAGE_ID="${{ github.actor }}/${IMAGE_NAME}"
-
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
-
-          echo "IMAGE_ID: ${IMAGE_ID}"
-          echo "VERSION: latest"
-
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:latest"
-          docker push "${IMAGE_ID}"
-
-      - name: Push release image
-        if: startsWith(github.ref, 'refs/tags/')
-        run: |
-          IMAGE_ID="${{ github.actor }}/${IMAGE_NAME}"
-
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
-
-          # Determine tag
-          TAG=$(git describe --exact-match --tags 2> /dev/null | sed -e s/v//g)
-          MAJOR=$(echo ${TAG} | cut -d'.' -f1)
-          FEATURE=${MAJOR}.$(echo ${TAG} | cut -d'.' -f2)
+      - name: Log into GitHub container registry
+        if: matrix.registry == 'github' && (startsWith(github.ref, 'refs/heads/master') || startsWith(github.ref, 'refs/heads/tags'))
+        run: echo "${{ secrets.GHCR_TOKEN }}" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
 
-          echo "IMAGE_ID: ${IMAGE_ID}"
-          echo "VERSION: ${TAG}"
+      - name: Determine image identifier for Docker Hub
+        if: matrix.registry == 'docker'
+        run: echo "IMAGE_ID=${{ github.actor }}/${{ env.IMAGE_NAME }}" >> $GITHUB_ENV
 
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:latest"
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:${TAG}"
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:${MAJOR}"
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:${FEATURE}"
-          docker push "${IMAGE_ID}"
+      - name: Determine image identifier for GitHub container registry
+        if: matrix.registry == 'github'
+        run: echo "IMAGE_ID=ghcr.io/${{ github.actor }}/${{ env.IMAGE_NAME }}" >> $GITHUB_ENV
 
-      - name: Log into github container registry
-        run: echo "${{ secrets.GHCR_TOKEN }}" | docker login ghcr.io -u "${{ github.actor }}" --password-stdin
+      - name: Normalize image identifier
+        run: echo "NORMALIZED_IMAGE_ID=$(echo ${{ env.IMAGE_ID}} | tr '[A-Z]' '[a-z]')" >> $GITHUB_ENV
 
       - name: Push latest image
+        if: startsWith(github.ref, 'refs/heads/master')
         run: |
-          IMAGE_ID="ghcr.io/${{ github.actor }}/${IMAGE_NAME}"
-
-          # Change all uppercase to lowercase
-          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
-
-          echo "IMAGE_ID: ${IMAGE_ID}"
+          echo "IMAGE_ID: ${{ env.NORMALIZED_IMAGE_ID }}"
           echo "VERSION: latest"
 
-          docker tag "${IMAGE_NAME}" "${IMAGE_ID}:latest"
-          docker push "${IMAGE_ID}"
+          docker tag "${{ env.IMAGE_NAME }}" "${{ env.NORMALIZED_IMAGE_ID }}:latest"
+          docker push "${{ env.MORMALIZED_IMAGE_ID }}"
```

There is a typo in `env.MORMALIZED_IMAGE_ID`

```suggestion
          docker push "${{ env.NORMALIZED_IMAGE_ID }}"
```